sso-kit-client/sso-kit-client-lit/src/AuthenticatedUser.ts

```diff
@@ -0,0 +1,120 @@
+/*-
+ * Copyright (C) 2022 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full
+ * license.
+ */
+import {makeAutoObservable} from "mobx";
+
+/**
+ * Type definition for the authenticated user information.
+ */
+export type User = {
+    birthdate?: string;
+    email?: string;
+    familyName?: string;
+    fullName?: string;
+    gender?: string;
+    givenName?: string;
+    locale?: string;
+    middleName?: string;
+    nickName?: string;
+    phoneNumber?: string;
+    picture?: string;
+    preferredUsername?: string;
+}
+
+/**
+ * A store for authenticated user..
+ */
+class AuthenticatedUser {
+
+    /**
+     * The user's birthdate.
+     */
+    birthdate?: string = undefined;
+    /**
+     * The user's email.
+     */
+    email?: string = undefined;
+    /**
+     * The user's family name.
+     */
+    familyName?: string = undefined;
+    /**
+     * The user's full name.
+     */
+    fullName?: string = undefined;
+    /**
+     * The user's gender.
+     */
+    gender?: string = undefined;
+    /**
+     * The user's given name.
+     */
+    givenName?: string = undefined;
+    /**
+     * The user's locale.
+     */
+    locale?: string = undefined;
+    /**
+     * The user's middle name.
+     */
+    middleName?: string = undefined;
+    /**
+     * The user's nickname.
+     */
+    nickName?: string = undefined;
+    /**
+     * The user's phone number.
+     */
+    phoneNumber?: string = undefined;
+    /**
+     * The user's picture.
+     */
+    picture?: string = undefined;
+    /**
+     * The user's preferred username.
+     */
+    preferredUsername?: string = undefined;
+
+    /**
+     * Constructor for the AuthenticatedUser.
+     */
+    constructor() {
+        makeAutoObservable(this);
+
+        // @ts-ignore: the imported file might not exist,
+        // in that case the authenticated user data will be empty
+        import("Frontend/generated/UserEndpoint").then((endpoint) => {
+                endpoint.getAuthenticatedUser().then(
+                    (user: User) => {
```

Comment:
`AuthenticatedUser` and `User` have the same fields. Looks like `User` is exported, but is not useful for the developer. You could maybe remove it and just use `AuthenticatedUser` as a type here.

Reply:
Actually, I have to change my comment here: I think that this implementation should be changed. Your class has a promise in the constructor, which means that when you use `new` on it, it will return an object that will have some values _after_ in time, but which can be used immediately. That can lead to errors.

I'd rather drop the `AuthenticatedUser` class altogether and export an `async` function that will eventually return the authenticated user, so that developers will have to adjust their code accordingly and `await`. That would also mean keeping the `User` type, of course.

Reply:
Removed the `AuthenticatedUser` class and ts file. Moved the User type in `SingleSignOnContext.ts`. Creating a user in `SingleSignOnContext` class.